django/contrib/auth/forms.py

```diff
@@ -111,6 +111,7 @@ def clean_password2(self):
     def save(self, commit=True):
         user = super(UserCreationForm, self).save(commit=False)
         user.set_password(self.cleaned_data["password1"])
+        user.clean()
```

Comment:
I don't expect this would be needed since `Model.clean()` is called as [part of `ModelForm` validation](https://docs.djangoproject.com/en/1.9/topics/forms/modelforms/#validation-on-a-modelform) (the test passes with this commented out). Was it just an oversight on your part?